cli/azd/internal/cmd/provision.go

```diff
@@ -199,7 +205,26 @@ func (p *ProvisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 	}
 	defer func() { _ = infra.Cleanup() }()
 
+	layer := ""
+	if len(p.args) > 0 {
+		layer = p.args[0]
+	}
+
 	infraOptions := infra.Options
+	if layer != "" {
+		layerOption, err := infraOptions.GetLayer(layer)
+		if err != nil {
+			return nil, err
+		}
+
+		infraOptions = layerOption
+	}
```

Comment:
If I have my infra set up like this, I currently get this error:

> ERROR: error executing step command 'provision layer1': layer 'layer1' not found in azure.yaml. available layers:

Seems like a base layer is required?

```
infra
├── abbreviations.json
├── layer-1
│   ├── main.bicep
│   └── main.parameters.json
├── layer-2
│   ├── main.bicep
│   └── main.parameters.json
├── layer-3
│   ├── main.bicep
│   └── main.parameters.json
└── resources
```

azure.yaml:

```yaml
infra:
  layers:
  - name: layer1
    path: ./infra/layer-1
  - name: layer2
    path: ./infra/layer-2
  - name: layer3
    path: ./infra/layer-3
workflows:
  up:
   steps:
    - azd: provision layer1
    - azd: provision layer2
    - azd: provision layer3
```

Reply:
I see. It's happening in this case where `infra/main.bicep` isn't present. The [ProjectImporter](https://github.com/Azure/azure-dev/blob/7c1e561f34252b39d8acd455cc5b48c3ba29503b/cli/azd/pkg/project/importer.go#L169) was returning a completely empty config in this case. This isn't quite as expected, and I've fixed the logic here.

Reply:
Thanks! With the new changes it detects the layers now